src/warehouse/util.js

```diff
@@ -46,12 +47,16 @@ const timestampRegex = new RegExp(
 );
 
 function validTimestamp(input) {
-  if (timestampRegex.test(input)) {
-    // check if date value lies in between min time and max time. if not then it's not a valid timestamp
-    const dateInMs = Date.parse(new Date(input).toISOString());
-    if (minTimeInMs <= dateInMs && dateInMs <= maxTimeInMs) {
-      return true;
+  try {
+    if (timestampRegex.test(input)) {
+      // check if date value lies in between min time and max time. if not then it's not a valid timestamp
+      const dateInMs = Date.parse(new Date(input).toISOString());
```

Comment:
I know this wasn't introduced here, but what's the reason we are doing:

`Date.parse(new Date(input).toISOString());` instead of `Date.parse(input)`

Reply:
Not sure of the reason. @Jayachand might have some more context.